pkg/serviceaccount/jwt.go

```diff
@@ -168,14 +169,28 @@ func (j *jwtTokenAuthenticator) AuthenticateToken(auds []string, tokenData strin
 		return nil, false, utilerrors.NewAggregate(errlist)
 	}
 
+	var (
+		aud string
+		ok  bool
+	)
+	if len(public.Audience) != 0 {
```

Comment:
don't we want to validate even when public.Audience is `""`, so that old service accounts don't pass validation for arbitrary audiences, only `""`? if we agree on the table in https://github.com/kubernetes/kubernetes/pull/62692#discussion_r192304746 as a goal, would be good to add test cases for as many of those as possible